fsm/leave_add_csync.go

```diff
@@ -207,6 +204,21 @@ func LeaveAddCsyncAction(z *music.Zone) bool {
 	}
 
 	updater := music.GetUpdater(leavingSigner.Method)
+	err, csyncrrs := updater.FetchRRset(leavingSigner, z.Name, z.Name, dns.TypeCSYNC)
+	if err != nil {
+		err, _ = z.SetStopReason(fmt.Sprintf("Unable to fetch CSYNC RRset from %s: %v", leavingSigner.Name, err))
+		return false
+	}
+	if len(csyncrrs) != 0 {
+		if err := updater.RemoveRRset(leavingSigner, z.Name, z.Name,
+			[][]dns.RR{[]dns.RR{z.CSYNC}}); err != nil {
+			z.SetStopReason(fmt.Sprintf("Unable to remove CSYNC record sets from %s: %s",
+				leavingSigner.Name, err))
+			return false
+		}
+		log.Printf("%s: Removed CSYNC record sets from %s successfully", z.Name, leavingSigner.Name)
```

Comment:
Consider using a log with different log levels to make it easier for the person looking at the logs to know what is important and what is not. Example on well used loggers are logrus and zap, there are probably more out there if you want something else.